python/dask_cudf/dask_cudf/_legacy/io/parquet.py

```diff
@@ -439,13 +439,12 @@ def set_object_dtypes_from_pa_schema(df, schema):
                 # However, this should be a simple range index that
                 # we can ignore anyway
                 continue
-            typ = cudf_dtype_from_pa_type(schema.field(col_name).type)
-            if (
-                col_name in schema.names
-                and not isinstance(typ, (cudf.ListDtype, cudf.StructDtype))
-                and isinstance(col, cudf.core.column.StringColumn)
-            ):
-                df._data[col_name] = col.astype(typ)
+            elif col_name in schema.names:
```

Comment:
I think the `if col_name is None:` branch above can be removed completely now?

Reply:
Yeah, we should be able to remove that block. Just added that change